.github/workflows/unit-tests.yml

```diff
@@ -23,21 +23,21 @@ jobs:
       matrix:
         include:
           # MacOSX
-          - { compiler: clang,    os: macos-12,     buildType: Debug, cxxStandard: 14 }
+          - { compiler: clang,    os: macos-12,     buildType: Debug }
           # Linux GCC
           # GCC 9 is known to show a few warnings that GCC 10 has "fixed", make sure this doesn't happen for us
-          - { compiler: gcc-9,    os: ubuntu-20.04, buildType: Debug }
-          - { compiler: gcc-10,   os: ubuntu-20.04, buildType: Debug, cxxStandard: 14, coverage: true }
-          - { compiler: gcc-10,   os: ubuntu-20.04, buildType: Debug, cxxStandard: 17 }
-          - { compiler: gcc-10,   os: ubuntu-20.04, buildType: Release, cxxStandard: 14 }
+          - { compiler: gcc-9,    os: ubuntu-20.04, buildType: Debug, boostVersion: 1.69.0 }
+          - { compiler: gcc-10,   os: ubuntu-20.04, buildType: Debug, boostVersion: 1.69.0, coverage: true }
+          - { compiler: gcc-10,   os: ubuntu-20.04, buildType: Release }
+          - { compiler: gcc-11,   os: ubuntu-22.04, buildType: Debug, boostVersion: 1.74.0 }
```

Comment:
I think it makes sense noting that GCC < 11 is C++14 and GCC >= 11 is C++17 at least as a comment so it is clear we test both. Similar for clang. You already gathered that info in the issue but that will be lost/hard to find when this is merged. Maybe even make it explicit by using `cxxStandard: 14` for the gcc-10 Release build and `cxxStandard: 17` for the gcc-12 release build.

Reply:
I wrote it down now in the comments